src/server-core/src/lib.rs

```diff
@@ -229,6 +230,50 @@ impl TlsCertConfig {
         builder.set_private_key_file(&self.key, SslFiletype::PEM)?;
         Ok(builder.build().into_context())
     }
+
+    /// Like [Self::context] but attempts to reload the files each time `ticker` yields an item.
+    /// Returns an error based on the files currently on disk. When `ticker` receives, the
+    /// certificates are reloaded from the context. The result of the reloading is returned on the
+    /// oneshot, and an Ok result means new connections will use the new certificates. An Err result
+    /// will not change the current certificates.
+    pub fn reloading_context(
+        &self,
+        mut ticker: BoxStream<'static, oneshot::Sender<Result<(), anyhow::Error>>>,
+    ) -> Result<ReloadingSslContext, anyhow::Error> {
+        let context = Arc::new(RwLock::new(self.context()?));
+        let updater_context = Arc::clone(&context);
+        let config = self.clone();
+        mz_ore::task::spawn(|| "TlsCertConfig reloading_context", async move {
+            while let Some(chan) = ticker.next().await {
+                let result = match config.context() {
```

Comment:
Initially I was thrown off a bit trying to figure out how reloading actually worked, I didn't realize that internally `fn context()` was reading new certs from a file. Would you mind renaming that method to `fn load_context()` to make it a bit more obvious?